packages/salesforcedx-vscode-core/src/commands/forceStartApexDebugLogging.ts

```diff
@@ -65,50 +78,69 @@ export class ForceStartApexDebugLoggingExecutor extends SfdxCommandletExecutor<{
           developerLogTraceFlag.validateDates();
-          updateTraceFlagCommandlet = new SfdxCommandlet(
-            new SfdxWorkspaceChecker(),
-            new EmptyParametersGatherer(),
-            new UpdateTraceFlagsExecutor()
-          );
-        }
-        if (updateTraceFlagCommandlet) {
-          promises.push(updateTraceFlagCommandlet.run());
+          execution = new CliCommandExecutor(
+            new UpdateTraceFlagsExecutor().build(),
+            {
+              cwd: vscode.workspace.rootPath
+            }
+          ).execute(cancellationToken);
+          this.attachSubExecution(execution);
+          resultPromise = new CommandOutput().getCmdResult(execution);
+          result = await resultPromise;
+          resultJson = JSON.parse(result);
         }
-
-        const updatePromise = Promise.all(promises);
-        await updatePromise;
       } else {
-        const createDebuglevelCommandlet = new SfdxCommandlet(
-          new SfdxWorkspaceChecker(),
-          new EmptyParametersGatherer(),
-          new CreateDebugLevel()
+        execution = new CliCommandExecutor(new CreateDebugLevel().build(), {
+          cwd: vscode.workspace.rootPath
+        }).execute(cancellationToken);
+        this.attachSubExecution(execution);
+        resultPromise = new CommandOutput().getCmdResult(execution);
+        result = await resultPromise;
+        resultJson = JSON.parse(result);
+        const debugLevelId = resultJson.result.id;
+        developerLogTraceFlag.setDebugLevelInfo(
+          debugLevelId,
+          APEX_CODE_DEBUG_LEVEL,
+          VISUALFORCE_DEBUG_LEVEL
         );
-        await createDebuglevelCommandlet.run();
+
         const userId = await getUserId(
           vscode.workspace.workspaceFolders![0].uri.fsPath
         );
-        const createTraceFlagCommandlet = new SfdxCommandlet(
-          new SfdxWorkspaceChecker(),
-          new EmptyParametersGatherer(),
-          new CreateTraceFlag(userId)
-        );
-        await createTraceFlagCommandlet.run();
+        developerLogTraceFlag.createTraceFlagInfo();
+        execution = new CliCommandExecutor(
+          new CreateTraceFlag(userId).build(),
+          {
+            cwd: vscode.workspace.rootPath
+          }
+        ).execute(cancellationToken);
+        this.attachSubExecution(execution);
+        resultPromise = new CommandOutput().getCmdResult(execution);
+        result = await resultPromise;
+        resultJson = JSON.parse(result);
+        developerLogTraceFlag.setTraceFlagId(resultJson.result.id);
       }
-      // run executors to update traceflag and debug levels
-      // tslint:disable-next-line:no-empty
-    } catch (e) {}
+      developerLogTraceFlag.turnOnLogging();
+      showTraceFlagExpiration(
+        developerLogTraceFlag.getExpirationDate().toLocaleString()
+      );
+      executionWrapper.successfulExit();
+    } catch (e) {
+      executionWrapper.failureExit(e);
```

Comment:
If a subexecution promise rejects, then it will be caught here and the top level wrapper will execute the failure method that emits a non 0 exit code.